src/librustc_hir/definitions.rs

```diff
@@ -461,14 +456,16 @@ impl Definitions {
         mapping: IndexVec<ast::NodeId, Option<hir::HirId>>,
     ) {
         assert!(
-            self.node_id_to_hir_id.is_empty(),
-            "trying to initialize `NodeId` -> `HirId` mapping twice"
+            self.def_id_to_hir_id.is_empty(),
+            "trying to initialize `LocalDefId` <-> `HirId` mappings twice"
         );
-        self.node_id_to_hir_id = mapping;
 
-        // Build the pre-computed mapping of `hir_id_to_node_id` -> `node_id_to_def_id`.
-        self.hir_id_to_def_id = self
-            .node_id_to_hir_id
+        for &node_id in self.def_id_to_node_id.iter() {
```

Comment:
Nit: why doesn't it use `collect`?

Reply:
For no reason actually. I'll make the change.

Reply:
Fixed